### PR TITLE
Add functional tests for using HttpContext

### PR DESCRIPTION
**Pull request by JamesNK**

### CODE REVIEW

File: testassets/FunctionalTestsWebsite/Greeter.cs [hunk 1]
```diff
@@ -39,6 +42,22 @@ public override Task<HelloReply> SayHello(HelloRequest request, ServerCallContex
         return Task.FromResult(new HelloReply { Message = "Hello " + request.Name });
     }
 
+    public override Task<HelloReply> SayHelloWithHttpContextAccessor(HelloRequest request, ServerCallContext context)
+    {
+        var httpContext = _httpContextAccessor.HttpContext;
```
Comment (davidfowl) [inline testassets/FunctionalTestsWebsite/Greeter.cs]: I hope this isn't the only way to get the HttpContext

Comment (JamesNK) [reply testassets/FunctionalTestsWebsite/Greeter.cs]: https://github.com/grpc/grpc-dotnet/pull/106/files#diff-e2b95dbc2cf449787b017fdd16ca0d35R55 ???

Comment (davidfowl) [reply testassets/FunctionalTestsWebsite/Greeter.cs]: Phew

Comment (davidfowl) [reply testassets/FunctionalTestsWebsite/Greeter.cs]: The SignalR similarities are uncanny

**Pull request merged**